File: src/Block/CreateTableBlock.java
/**
 * This file copyright (c) 2015-2016, Jing Fan, Adalbert Gerald Soosai Raj, and Jignesh M. Patel
 *
 * See the file CREDITS.txt in the root directory for details.
 **/

package Block;

/**
 * @brief The block of SQL to create a table.
 *
 */
public class CreateTableBlock extends Block {

  /**
   * @brief Constructor.
   * @param stage What's this block for.
   * @param indent Indentation level.
   * @param tableName The new table name.
   * @param attrs The attributes, every attr should contain full description
   * including attribute name, type.
   */
  public CreateTableBlock(String stage,
                          int indent,
                          String tableName,
                          String[] attrs) {
    super(stage, indent);
    this.append("CREATE TABLE " + tableName + "(");
    for (String attr : attrs) {
      this.append(attr + ",", indent + 1);
    }
    this.append(")" + ";");
    this.sql = this.sb.toString();
  }

}

File: python/src/Blocks/CreateTableBlock.py
from Block import Block
class CreateTableBlock(Block):
    '''
    generates the create table sql statement
    '''
    
    def __init__(self,stage,indent,tableName,attrs):
        '''
        constructor: generates a create table sql
        @param stage:current stage
        @type stage: string
        @param indent: indent level
        @type indent: int
        @param tableName: Name of table to create
        @type tableName: string
        @param attrs: attribute list to have in the table
        @type attrs: list
        '''
        super(CreateTableBlock,self).__init__(stage,indent)
        self.append("CREATE TABLE " + tableName + "(")
        ind=0
        for attr in attrs:
            if(ind==len(attrs)-1):
                self.append(attr,indent+1)
            else:    
                self.append(attr+",",indent+1)  
            ind=ind+1      
        self.append(")" + ";")   
        self.sql=self.sb 